File: app/src/main/java/com/telolahy/towerofhanoi/scene/SplashScene.java
package com.telolahy.towerofhanoi.scene;

import org.andengine.entity.sprite.Sprite;

/**
 * Created by stephanohuguestelolahy on 11/4/14.
 */
public class SplashScene extends BaseScene {

    private Sprite mBackground;

    @Override
    public void createScene() {

        mBackground = new Sprite(400, 240, mResourcesManager.splashTextureRegion, mVertexBufferObjectManager);
        attachChild(mBackground);
    }

    @Override
    public void onBackKeyPressed() {

    }

    @Override
    public void disposeScene() {

        mBackground.detachSelf();
        mBackground.dispose();
        this.detachSelf();
        this.dispose();

    }
}

File: app/src/main/java/com/telolahy/towerofhanoi/scene/SplashScene.kt
package com.telolahy.towerofhanoi.scene

import com.telolahy.towerofhanoi.Constants
import org.andengine.entity.sprite.Sprite

/**
 * Created by stephanohuguestelolahy on 11/4/14.
 */
class SplashScene : BaseScene() {
  private var mBackground: Sprite? = null

  override fun createScene() {
    mBackground = Sprite(Constants.CAMERA_WIDTH / 2, Constants.CAMERA_HEIGHT / 2, mResourcesManager.splashTexture.textureRegion, mVertexBufferObjectManager)
    attachChild(mBackground)
  }

  override fun onBackKeyPressed() {}

  override fun disposeScene() {
    mBackground?.detachSelf()
    mBackground?.dispose()
    detachSelf()
    dispose()
  }
}